x-pack/plugins/rule_registry/common/types.ts

```diff
@@ -311,6 +311,73 @@ export const bucketAggsSchemas = t.intersection([
   ),
 ]);
 
+export const alertsAggregationsSchema = t.record(
+  t.string,
+  t.intersection([metricsAggsSchemas, bucketAggsSchemas])
+);
+
+export const alertsGroupFilterSchema = t.record(
+  t.union([
```

Comment:
Do we have existing enums for these literals? Just so we can have a more centralized source of truth

Reply:
Would have been nice, but I was only able to find [type-level information](https://github.com/elastic/elasticsearch-js/blob/94bf5b2aa7f6a2eb3e310a26ea10d6ff91c467c4/src/api/typesWithBodyKey.ts#L5917). I added a type assertion to check that the keys are correctly aligned to the es types though 🙂